referencing/faq.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: default
+title: Referencing FAQ
+nav_order: 1
+parent: Using References
+---
+
+# Referencing FAQ
+
+## Do I need to read the entire referencing guide?
+
+If your OpenAPI Description (OAD):
+
+1. consists only of a single JSON or YAML document
+1. contains only references that are fragments (beginning with a `"#"`, e.g. `{"$ref": "#/components/responses/oops"}`)
+1. does not have any other keywords in the same object as any `"$ref"`
+1. (OAS 3.1 only) does *not* use `"$id"`if you are using OAS 3.1, you are  *not* using the `"$id"` keyword in Schema Objects
+1. (OAS 3.1 only) does *not* use `"$dynamicRef"` or `"$dynamicAnchor"`
+
+Then your use of references is simple enough that your references will do what you expect with any tool that supports them at all.  You probably do not need to read the rest of this page.
+
+## Can an OAD be *completely* "de-referenced" by "removing" all references?
+
+In general, no.  But in many common cases, yes, and tools exist that will attempt this.
+
+There are two types of reasons why de-referencing might not be possible:
+
+* Technical reasons:
+   * Cyclic references are allowed in Schema Objects, and are used to describe recursive tree structures.  JSON Schema itself is a recursive tree structure, so the JSON Schema metaschemas cannot be fully de-referenced (see draft-07 or earlier for how this works with `"$ref"`; draft 2020-12 uses a different technique)
+   * Cyclic references can also occur when describing complex payloads such as `multipart/form-data` where media type descriptions can be nested
+   * References in Link Objects (`"operationRef"`) and Discriminator Objects (`"mapping"`) cannot be replaced with literal values, although there are alternatives to using references in these cases (`"operationId"` and mapping by schema component name, respectively, althouogh there can be challenges with arranging the documents properly for those techniques as well)
+   * Dynamic references in OAS 3.1 (which are not covered in detail by this guide) have runtime behavior which cannot always be predicted ahead of time
+* Social reasons:
```

Comment:
I didn't totally understand the objections in today's discussion, but I felt perhaps there was a desire to acknowledge these concerns, while not putting them front-and-center.

Rather than a rich breakdown of each of these *valid* social concerns, an extremely terse caveat could cover the same terrain from a slightly-higher elevation.

```suggestion
* Social reasons:
  Complex OpenAPI Descriptions may have many stakeholders, and be deployed and consumed in many contexts. Authors of OpenAPI Descriptions may need to consider these contexts when de-referencing.
  * Different owners; parts of an API may be owned by different teams, and may be deployed on different schedules
  * Security Policies; parts of an API may not be accessible for de-referencing by all consumers
```

In the service of brevity, I sort of squashed-together the `Large-scale` and `Independent Deployment` bits of the original bullet points.

The `Standards Compliance` piece is the bullet I understand the least (in terms of how an OAD might need to be split-up), but if there's a way to describe it succinctly, or to roll it into one of the other bullets, that seems like a win.

Again, I'm just trying to respond to what felt like a desire to downplay the (over)-articulation of these caveats, and to highlight the *Happy Path* for casual readers of these documents.

The concerns seem valid, but those who are affected by those concerns will likely have their ears pricked by the presence of the keywords, and don't need their concerns articulated in this context.

Reply:
I like these suggestions. 

My general point in these discussions has been to observe that in a compile-time contract context, it might be helpful to publish a version of the description that is optimized for the consumer's point of view. In this case, you might consider doing things like merging multiple documents into a single one, and/or you might exclude operations to which the consumer was not entitled. In this way, the points about different owners and different security polices would be opaque (dynamic, just-in-time contract probably wouldn't ever fit this simplest of contexts).